stan/math/prim/arr/err/is_nonzero_size.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef STAN_MATH_PRIM_SCAL_ERR_IS_NONZERO_SIZE_HPP
+#define STAN_MATH_PRIM_SCAL_ERR_IS_NONZERO_SIZE_HPP
+
+namespace stan {
+namespace math {
+
+/**
+ * Check is the specified matrix/vector is on non-zero size
+ *
+ * @tparam T_y Type of container
```

Comment:
Similar to the comment above: we should mention that this needs to follow a concept... that it needs a `.size()` function defined.